src/app/repositories/tasks.py

```diff
@@ -1,45 +1,62 @@
-from abc import ABC, abstractmethod  # noqa
+from abc import ABC, abstractmethod
 from uuid import UUID
 
-from sqlalchemy import select  # noqa
-from sqlalchemy.ext.asyncio import AsyncSession  # noqa
+from sqlalchemy import select
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm import contains_eager
 
-from app.schemas.tasks import TaskSchema  # noqa
-from database.models.task import Task  # noqa
+from app.schemas.tasks import TaskCreateSchema, TaskGetSchema, TaskUpdateSchema, TaskWithStatusSchema
+from database.models.task import Task
 
 
 class AbstractTaskRepository(ABC):
     @abstractmethod
-    async def get(self, obj_id: UUID) -> TaskSchema:
+    async def get(self, obj_id: UUID) -> TaskGetSchema:
         raise NotImplementedError
 
     @abstractmethod
-    async def put(self, obj_in: TaskSchema, **kwargs: dict) -> TaskSchema:
+    async def create(self, obj_in: TaskCreateSchema, **kwargs: dict) -> TaskGetSchema:
         raise NotImplementedError
 
     @abstractmethod
-    async def delete(self, db_obj: TaskSchema) -> TaskSchema:
+    async def update(self, obj_id: UUID, obj_in: TaskUpdateSchema) -> TaskGetSchema:
         raise NotImplementedError
 
+    @abstractmethod
+    async def delete(self, obj_id: UUID) -> dict:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def get_all_by_idp_id_with_status(self, idp_id: UUID) -> list[TaskWithStatusSchema]:
+        raise NotImplementedError
 
-class SqlAlchemyTaskRepository(AbstractTaskRepository):
-    def __init__(self, session: AsyncSession, model: Task = Task) -> None:
+
+class SQLAlchemyTaskRepository(AbstractTaskRepository):
+    def __init__(self, session: AsyncSession, model: type[Task] = Task) -> None:
         self._session = session
         self._model = model
 
-    async def get(self, obj_id: UUID) -> TaskSchema:
+    async def get(self, obj_id: UUID) -> TaskGetSchema:
         obj = await self._session.execute(select(self._model).where(self._model.id == obj_id))
         return obj.scalars().first()
```

Comment:
Ты возвращаешь объект User, а нужно TaskGetSchema
```suggestion
    obj = obj.scalars().first()
    return TaskGetSchema.model_validate(obj)
```

Проверь что у TaskGetSchema настроен параметр from_attributes=True. Либо можешь его передал в вызов функции model_validate(obj, from_attributes=True)

Это нужно для инверсии зависимости (буква D в SOLID). У нас service не должен зависить никак от того по какой логике работает репозиторий. Если он будет возвращать модель базы данных, то это утечка зависимости. Если в альфа банк будут использовать не алхимию и не постгрес, а что-то другое, то у них будут проблемы когда они захотят изменить логику работы с базой. Чтобы этого избежать мы и используем инверсию зависимости с помощью `AbstractTaskRepository` и `TaskGetSchema`.

Понимаю, что это может немного сложно для понимания. Сам долго разбиралс в этом принципе

Reply:
Соответственно в других местах то же самое нужно сделать

Reply:
Плюс нужно сделать проверку на то что запрос тебе что-то вернул. Если нет, то нужно зарейзить ошибку.